zips/zip-0228.rst

```diff
@@ -2,12 +2,539 @@
+-------------------------
+
+In this ZIP, **we assume that orders are matched perfectly**, and we do not support partial fills.
+In practice though, modern CLOBs offer partial fills to their users, allowing trades with matching prices to be matched even if quantities don't align perfectly.
+Offering partial fills without making strong assumptions on the Matcher (e.g. “The Matcher is a market maker that takes the other side of each trade”) or on the availability of traders (e.g. “Traders stay online and stay connected to the Matcher to split their orders on-demand”) is hard.
+This is particularly at odds with the protocol in this ZIP, since the Matcher is non-custodian and thus only manipulates commitments and notes, which are generated before the matching occurs: traders commit to values, which may need to be changed - in the future - to align with the matching operation of the Matcher (done at a subsequent time).
+
+Some of the proposals to allow for this are:
+
+- The use of standard denomination for orders.
+- The inclusion of a special flag “partialFill = true” to flag to the Matcher that the trader is willing to stay online to split orders on demand.
+- Existence of a third party Market Maker providing liquidity to the many ZSA pairs traded on the system.
+
+All of these proposals have their pros and cons, and warrant further study and discussion.
+
+Fairness
+--------
+
+It is important to provide a good execution for both sides of a Swap. 
+As such, one side should not have an advantage against the other. 
+Every trading party must be able to withdraw their trading intent (i.e., cancel their orders).
+The main considerations here are:
+
+- Provide a mechanism for time-limiting Swaps to reduce freezing up of Assets due to their being locked up in Swaps that have not been completed, and to align with dynamically changing “exchange rates” between different Assets.
+- Consider repercussions of, and possible mitigations for, the “free option” that becomes available to the party that performs the last step of the protocol. That is, it is possible for that party to abort the protocol at this last step if the Swap is no longer beneficial for it.
+
+Denial-of-Service (DoS) Prevention
+----------------------------------
+
+The implementation of a fee mechanism is not enough to provide DOS prevention. 
+In fact, the presence of a fee in a Swap Order doesn't ensure this fee is payable. 
+A trader, Alice, can promise to pay 1000 ZEC, but this promise has no value unless it can be verified. The same applies to fees. 
+The Matcher must be able to carry out quick validity checks on Swap Orders to assess their probability to form valid Swap Bundles when matched. 
+This includes, verifying the ability for traders to pay the fees they must pay as part of the protocol. 
+These checks are probabilistic (the state of the blockchain can change in the meantime), but provide some guarantees to the Matcher that it will be paid for its services.
```

Comment:
`probabilistic checks` means something else. simply `these checks provide some guarantees to the Matcher that the components of the Swap Order are valid`

Reply:
Done.